internal/configs/test_file.go

```diff
@@ -396,10 +397,22 @@ func loadTestFile(body hcl.Body) (*TestFile, hcl.Diagnostics) {
 			}
 
 		case "provider":
-			provider, providerDiags := decodeProviderBlock(block)
+			providerBlock, providerDiags := decodeProviderBlock(block)
 			diags = append(diags, providerDiags...)
-			if provider != nil {
-				tf.Providers[provider.moduleUniqueKey()] = provider
+			if providerBlock != nil {
+				_, decodeDiags := providerBlock.decodeStaticFields(nil)
+				diags = append(diags, decodeDiags...)
+				if decodeDiags.HasErrors() {
+					continue
+				}
+				providers, diagsStatic := providerBlock.decodeStaticFields(nil)
+				diags = append(diags, diagsStatic...)
+				if diagsStatic.HasErrors() {
+					continue
+				}
+				for _, provider := range providers {
```

Comment:
Creating a provider list out of a provider block with a nil static eval should always result in a single provider, right?

nit: should we just make it `providers[0]` for clarity?

Reply:
I asked to keep it as a loop to future proof it.